### DISCUSSION TITLE
Please provide an example of monitoring a Java application

### DISCUSSION THREAD

**Discussion by DerickIT** [ccfos/nightingale#1181] [Q&A]
http://n9e.flashcat.cloud/docs/usage/apm/   是需要应用集成agent上报，还是通过什么方式，有哪些参考案例

**Comment by UlricQin** [comment]
文档中的例子，是演示使用SDK埋点的方式来采集监控数据。

Java的话也可以用这种SDK埋点的方式，Java也有Prometheus的SDK，也可以用 statsd 埋点，当然，这种方式有些人会觉得侵入性比较高，但是灵活性好，因为可以埋点采集一些业务指标，如果监控做的精细化，一定是需要采集业务指标的。

Java还有一些其他的埋点工具，比如SpringBoot的程序，可以直接使用 actuator，actuator底层是使用的 micrometer，所以如果是非SpringBoot程序，也可以直接使用micrometer。

或者，使用javaagent的方式，比如使用 jmx_exporter：https://github.com/prometheus/jmx_exporter
